src/igr-tests/igr_functions

```diff
@@ -0,0 +1,232 @@
+# POSIX documented things.
+# So we replace shell built-in echo with a printf based function.
+# For more info see http://www.etalabs.net/sh_tricks.html
+echo() {
+    IFS=" " printf %s\\n "$*"
+}
+error() {
+    >&2 echo "${TEST_NAME:-}: Error: $*"
+    # Built-in sub_error
+    if [ -n "${2:-}" ]; then
+        >&2 echo " ... $2"
+    fi
+    if [ -n "${STAGE:-}" ]; then
+        echo "${TEST_NAME:-}: Failed at stage $STAGE."
+    fi
+    if [ -e "${SOCKET:-}" ]; then
+        stop_dinit # A dinit instance is running, stopping...
+    fi
+    exit 1
+}
+warning() {
+    echo
+    >&2 echo "${TEST_NAME:-}: Warning: $1"
+    # Built-in sub_warning
+    if [ -n "${2:-}" ]; then
+        >&2 echo " ... $2"
+    fi
+    echo
+}
+
+## Executable path resolvers functions
+# These functions return 0 and set suitable variable when program found.
+# And return 1 on failure.
+__find_dinit() {
+    if [ -z "${DINIT:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinit" ]; then
+            export DINIT="$EXEC_PREFIX/dinit"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+__find_dinitctl() {
+    if [ -z "${DINITCTL:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinitctl" ]; then
+            export DINITCTL="$EXEC_PREFIX/dinitctl"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+__find_dinitcheck() {
+    if [ -z "${DINITCHECK:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinitcheck" ]; then
+            export DINITCHECK="$EXEC_PREFIX/dinitcheck"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+__find_dinitmonitor() {
+    if [ -z "${DINITMONITOR:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinit-monitor" ]; then
+            export DINITMONITOR="$EXEC_PREFIX/dinit-monitor"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+
+## Change directory to run-test.sh location
+cd "$(dirname "$0")" || error "Can't change directory to script location."
+
+## Prepare $IGR_OUTPUT
+TEST_NAME="${PWD##*/}"
+if [ -z "$TEST_NAME" ]; then error "Failed to guess test name."; fi
+if [ -n "${MAKE:-}" ]; then
+    IGR_OUTPUT="$PWD"
+    if [ -z "$IGR_OUTPUT" ]; then error "Failed to guess igr output files location."; fi
+    export IGR_OUTPUT
+else
+    # Igr tests are called by meson.
+    mkdir -p "$IGR_OUTPUT/$TEST_NAME/"
+    IGR_OUTPUT="$IGR_OUTPUT/$TEST_NAME/"
+    export IGR_OUTPUT
+fi
+mkdir -p "$IGR_OUTPUT"/output/
+export SOCKET="$IGR_OUTPUT/socket"
+if [ -n "${DEBUG:-}" ]; then
+    QUIET=""
+else
+    QUIET="--quiet"
+fi
+
+## Integreation tests helper functions
+# This function spawn a dinit daemon with temp socket.
+# It's pass its flags as dinit daemon flags.
+# RESULT: return 0 & sets "SOCKET" & "DINITPID" variables on sccuess.
+#         throw an error on failure.
+spawn_dinit() {
+    __find_dinit || error "Cannot find dinit exec path." "Check 'EXEC_PREFIX' variable value"
+    "$DINIT" $QUIET -u -d sd -p "$SOCKET" "$@" &
```

Comment:
Here it uses "$SOCKET" but the comment says that the function _sets_ SOCKET

Reply:
You've marked this as resolved, but I don't see the change. The comment above this function still says that it sets SOCKET, but it doesn't set SOCKET.

Reply:
Still need to resolve this properly

Reply:
I mention it's spawn dinit on "$SOCKET" socket path instead of setting "$SOCKET".